x-pack/plugins/observability_solution/inventory/public/components/entities_grid/index.tsx

```diff
@@ -136,6 +154,28 @@ export function EntitiesGrid({
     [onChangeSort]
   );
 
+  const getEntityRedirectUrl = useCallback(
+    (entity: LatestEntity) => {
+      const type = entity[ENTITY_TYPE] as EntityType;
+
+      // Any unrecognised types will always return undefined
+      switch (type) {
+        case 'host':
+        case 'container':
+          return assetDetailsLocator?.getRedirectUrl({
+            assetId: entity[ENTITY_DISPLAY_NAME],
+            assetType: type,
+          });
+
+        case 'service':
+          return serviceOverviewLocator?.getRedirectUrl({
+            serviceName: entity[ENTITY_DISPLAY_NAME],
```

Comment:
In most of the cases, this won't work. This is because the `ENTITY_DISPLAY_NAME` field is a joins the service.name and service.environment like `service.name:service.environment`. So if you pass it APM will never find it.